src/index-manager.ts

```diff
@@ -10,6 +10,17 @@ export class IndexManager {
 	 */
 	constructor(private readonly collection: Collection) {}
 
+	/**
+	 * Returns a list of all indexes.
+	 */
+	public async findAllIndexes(): Promise<IndexSpecification[]> {
+		try {
+			return await this.collection.listIndexes().toArray();
```

Comment:
:warning: confusion in promises 

```ts
await (await this.collection.listIndexes()).toArray()
```

Reply:
Is the inner await really necessary in this situation?